src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -30,9 +31,159 @@ namespace {
 
 namespace query = xrt_core::query;
 using key_type = query::key_type;
+xclDeviceHandle handle;
+xclDeviceInfo2 deviceInfo;
 
 static std::map<query::key_type, std::unique_ptr<query::request>> query_tbl;
 
+static zynq_device*
+get_edgedev(const xrt_core::device* device)
+{
+  return zynq_device::get_dev();
+}
+
+struct devInfo
+{
+   static boost::any
+    get(const xrt_core::device* device,key_type key)
+  {
+    auto edev = get_edgedev(device);
+    switch (key)
+    {
+    case key_type::edge_vendor:
+      return deviceInfo.mVendorId;
+    case key_type::rom_vbnv:
+      return std::string(deviceInfo.mName);
+    case key_type::rom_ddr_bank_size:
+      return (deviceInfo.mDDRSize >> 30);
+    case key_type::rom_ddr_bank_count_max:
+      return static_cast<uint64_t>(deviceInfo.mDDRBankCount);
+    case key_type::clock_freqs:
+      {
+        std::vector<std::string> clk_freqs;
+        for(int i = 0; i < 3; i++)
```

Comment:
Magic numbers.  What does '3' represent.  You might wish to consider determining the size directly from the structure.  For example: sizeof(xclDeviceInfo2.mOCLFrequency)